src/components/Dropdown/DropdownList.tsx

```diff
@@ -56,21 +57,98 @@ export const DropdownList = (props: IDropdownListProps): ReactElement => {
     ],
   });
 
+  const close = () => {
+    if (visible) {
+      setVisible(false);
+      onClosed();
+    }
+  };
+
+  const open = () => {
+    if (!visible) {
+      setVisible(true);
+      onExpanded();
+    }
+  };
+
+  /* Click on dropdown */
   const handleClick = () => {
     setVisible(!visible);
+    visible ? onExpanded() : onClosed();
   };
 
-  const handleSelect = useCallback(
-    (item: ItemType) => {
-      return () => {
-        onSelect(item.id);
-      };
-    },
-    [onSelect],
-  );
+  /* keydown on dropdown */
+  const handleKeyDown = (e: KeyboardEvent) => {
```

Comment:
I mentioned the same below, did a similar update here:

```js
const handleKeyDown = (e: KeyboardEvent) => {
  switch (e.key) {
    case 'Enter':
    case 'Space':
      e.preventDefault();
      return handleClick();
    case 'Escape':
      return close();
    case 'ArrowDown':
      e.preventDefault();
      open();
      return focusItem(0);
  }
};
```

Reply:
Updated